lib/matplotlib/collections.py

```diff
@@ -906,8 +906,11 @@ def update_scalarmappable(self):
         """
         if not self._set_mappable_flags():
             return
+        # We don't use the result of this call, but update the status for
+        # anyone watching.
+        self._check_update("array")
         # Allow possibility to call 'self.set_array(None)'.
-        if self._check_update("array") and self._A is not None:
+        if self._A is not None:
```

Comment:
The issue is that this was coming back false, but we should have gone through this branch?